firebase-messaging/src/main/java/com/google/firebase/messaging/ImageDownload.java

```diff
@@ -115,9 +115,16 @@ public Bitmap blockingDownload() throws IOException {
     try (InputStream connectionInputStream = connection.getInputStream()) {
       // Read one byte over the limit so we can tell if the data is too big, as in many cases
       // BitmapFactory will happily decode a partial image.
-      bytes =
-          ByteStreams.toByteArray(
-              ByteStreams.limit(connectionInputStream, MAX_IMAGE_SIZE_BYTES + 1));
+      ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+
+      int nRead;
+      byte[] data = new byte[MAX_IMAGE_SIZE_BYTES + 1];
+
+      while ((nRead = connectionInputStream.read(data, 0, data.length)) != -1) {
+        buffer.write(data, 0, nRead);
```

Comment:
This can end up reading several chunks of *up to* `MAX_IMAGE_SIZE_BYTES+1`, which is not equivalent to `ByteStreams.limit`. If you check @ [ByteStreams implementation](https://github.com/google/guava/blob/master/guava/src/com/google/common/io/ByteStreams.java#L744) it limits the *total* value read to be up to `MAX_IMAGE_SIZE_BYTES+1` with a counter. You would need to implement something similar.

Reply:
i think it will be equivalent. bcz the reading is not in the for loop and ByteStreams.limit(connectionInputStream, MAX_IMAGE_SIZE_BYTES + 1)); is getting initialized everytime a function is called so the limit will be set everytime not once